x-pack/test/api_integration/apis/entity_manager/helpers/request.ts

```diff
@@ -30,9 +30,9 @@ export const installDefinition = async (supertest: Agent, definition: EntityDefi
     .expect(200);
 };
 
-export const uninstallDefinition = (supertest: Agent, id: string) => {
+export const uninstallDefinition = (supertest: Agent, id: string, deleteData = false) => {
   return supertest
-    .delete(`/internal/entities/definition/${id}`)
+    .delete(`/internal/entities/definition/${id}${deleteData ? '?deleteData=true' : ''}`)
```

Comment:
nit you can pass a [query object](https://github.com/elastic/kibana/blob/main/x-pack/test/api_integration/apis/entity_manager/enablement.ts#L24) to the request which could be cleaner

Reply:
Done!